src/app/api/dashboard/[type]/route.js

```diff
@@ -85,21 +89,22 @@ export async function GET(req, { params }) {
           where(`roles.${params.type}`, "in", [-1, 0, 1])
         )
       );
-      snapshot.forEach((doc) => {
+      const promises = snapshot.docs.map(async (doc) => {
```

Comment:
get doesnt need it either since its a single read operation. the for loop here is not something we will be able to optimize